Posts/2021/07/How Can I Be Notified Any Time a Service Goes Down.md

```diff
@@ -0,0 +1,209 @@
+
+```powershell-console
+PS C:\> Get-CimInstance -Namespace root\CIMV2 -ClassName win32_service
+```
+
+The result looks like the following.
+
+```powershell-console
+PS C:\> Get-CimInstance -Namespace root\CIMV2 -ClassName win32_service
+```
+
+```powershell
+PS C:\> Get-CimInstance -Namespace root\CIMV2 -ClassName win32_service
+
+ProcessId Name                                     StartMode State   Status  ExitCode
+--------- ----                                     --------- -----   ------  --------
+3784      AdobeARMservice                          Auto      Running OK      0
+3792      AdobeUpdateService                       Auto      Running OK      0
+3800      AGMService                               Auto      Running OK      0
+3824      AGSService                               Auto      Running OK      0
+0         AJRouter                                 Manual    Stopped OK      1077
+0         ALG                                      Manual    Stopped OK      1077
+0         AppIDSvc                                 Manual    Stopped OK      1077
+6708      Appinfo                                  Manual    Running OK      0
+21444     AppMgmt                                  Manual    Running OK      0
+0         AppReadiness                             Manual    Stopped OK      1077
+0         AppVClient                               Disabled  Stopped OK      1077
+0         AppXSvc                                  Manual    Stopped OK      0
+0         AssignedAccessManagerSvc                 Manual    Stopped OK      1077
+```
+
+For now, you can use PowerShell to find the required class, and enumerate it to get the available services.
+Let's go deeper now and start creating the WMI Event Subscription.
+
+There are three steps you need to follow to create a WMI Event Subscription:
+
+- Create the WMI query
+- Registering the query
+- Reading the events.
+
+## Creating WMI Query
+
+WMI database has several system WMI Classes.
+For instance, the *CIM_InstModification* monitors the targeted class for any changes.
+So, in this case, *Win32_Service*.
+
+The query syntax looks like
+
+```powershell-console
+Select * from <WMI System Class> within <Number of Seconds> where TargetInstance ISA <WMI Class name>
+```
+
+Let apply the same to *Win32_Serivce.* Start by Creating a PowerShell variable name it $Query and type the following query
+
+```powershell-console
+PS C:\> $query = "Select * from CIM_InstModification within 10 where TargetInstance ISA 'Win32_Service'"
+```
+
+> A full explanation for the WQL query is available in [Your Goto Guide for Working with Windows WMI Events and PowerShell](https://adamtheautomator.com/your-goto-guide-for-working-with-windows-wmi-events-and-powershell/)
+
+## Registering The Query
+
+We have the WQL query, let move to the next step and register the query to the WMI events by using the **Register-CimIndicationEvent**

```

Comment:
Explain what this command does

Reply:
Adding a link to it on Microsoft site, is it ok